### PR TITLE
The Security Techfab now produces basic ammo

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
tweak: The Security Techfab can now produce basic shotgun shell types and .38 ammo
/:cl:

.5 hours of testing so we're good to go boyyysss

### CODE REVIEW

**Comment by 81Denton** [comment]
Good change, I was always wondering why the autolathe can print ammo but the security lathe couldn't.

**Comment by 81Denton** [comment]
Good change, I was always wondering why the autolathe could print ammo yet the security lathe couldn't.

**Comment by Robustin** [comment]
Good change, I was always wondering why the autolathe could print ammo yet the security lathe couldn't.

**Comment by AlexTheSysop42** [comment]
Good change, I was always wondering why the autolathe could print ammo yet the security lathe couldn't.

**Comment by Mark9013100** [comment]
Good change, I was always wondering why the autolathe could print ammo yet the security lathe couldn't.

**Comment by 81Denton** [comment]
yeah same really

**Comment by 81Denton** [comment]
mad mads

**Comment by ExcessiveUseOfCobblestone** [comment]
Good change, I was always wondering why the autolathe could print ammo yet the security lathe couldn't.

**Comment by ExcessiveUseOfCobblestone** [comment]
all you did was add a bit and relocated them, it should work.

**Comment by CPTANT** [comment]
Hey why does this remove the ability to make them in the autolathe?

Undeclared change and not a good one, there should be more places to print ammo than just in the brig.

**Comment by Robustin** [comment]
It doesn't I was just clarifying in the commit names that designs were being moved to another file.

That seems to be the standard practice for "hybrid" designs.

**Comment by CPTANT** [comment]
How does the "hacked" requirement work with the departmental lathe?

**Comment by ExcessiveUseOfCobblestone** [comment]
it doesn't check for it.

**Comment by Robustin** [comment]
.1 hours of testing done

This apparently lets the autolathe produce this ammo by default, ruh roh

**Comment by Bawhoppen** [comment]
I feel like this devalues interactions with cargo even more.

**Comment by Robustin** [comment]
@Bawhoppen 

Box is the only station that doesnt have a public autolathe and (hopefully) that will change soon too. There's no interaction for ammo as it is.

**Comment by YakumoChen** [comment]
does it let you print .357 and lethal ammo in the techfab?

**Comment by ExcessiveUseOfCobblestone** [comment]
@Robustin https://github.com/tgstation/tgstation/blob/28b1fb254d23678fb4c9787471ef851bb11a5c8d/code/game/machinery/autolathe.dm#L378

Dummy put TRUE instead of FALSE so it gives the hacked designs on initialize.

**Comment by Robustin** [comment]
@ExcessiveUseOfCobblestone 

I'm confused, that line you linked is just for the HACKED subtype of autolathes - which I don't even know where we use those.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Robustin im a chumbis.

Try calling adjust_hacked(0) when you spawn it in and tell me what happens~

**Comment by Robustin** [comment]
Spawn what in? The autolathe? It seems like the cleanest solution is just to have 2 design schematics for this ammo until someone sorts out getting dual-designs to work with hacked items.

**Comment by ExcessiveUseOfCobblestone** [comment]
Yes the autolathe.

Having 2 design schematics is a bandaid for the system not working. It is the EASIEST, not the cleanest.

**Comment by Robustin** [comment]
I dont think manually unhacking the autolathe during initialization is a clean or effective solution either.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Robustin I was asking you to test not to pr reeeeee

**Comment by Robustin** [comment]
[15:40:49] Runtime in unsorted.dm,1254: WARNING: Design ID clash with ID rubber_shot detected!

Jesus christ its like this system is designed not to work

**Comment by BeeSting12** [comment]
Good change, I was always wondering why the autolathe could print ammo yet the security lathe couldn't.

File: code/modules/research/designs/weapon_designs.dm [hunk 1]
```diff
@@ -2,6 +2,60 @@
...
 /////////////////////////////////////////
 
+/datum/design/c38/sec
+	name = "Speed Loader (.38)"
+	id = "sec_38"
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 20000)
+	build_path = /obj/item/ammo_box/c38
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_SECURITY
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/designs/weapon_designs.dm]: Remove 6,7,9,10

File: code/modules/research/designs/weapon_designs.dm [hunk 2]
```diff
@@ -2,6 +2,60 @@
...
+
+/datum/design/rubbershot/sec
+	name = "Rubber Shot"
+	id = "sec_rshot"
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000)
+	build_path = /obj/item/ammo_casing/shotgun/rubbershot
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_SECURITY
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/designs/weapon_designs.dm]: 15,16,18,19

File: code/modules/research/designs/weapon_designs.dm [hunk 3]
```diff
@@ -2,6 +2,60 @@
...
+
+/datum/design/shotgun_slug/sec
+	name = "Shotgun Slug"
+	id = "sec_slug"
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000)
+	build_path = /obj/item/ammo_casing/shotgun
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_SECURITY
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/designs/weapon_designs.dm]: 24,25,27,28

File: code/modules/research/designs/weapon_designs.dm [hunk 4]
```diff
@@ -2,6 +2,60 @@
...
+
+/datum/design/buckshot_shell/sec
+	name = "Buckshot Shell"
+	id = "sec_bshot"
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000)
+	build_path = /obj/item/ammo_casing/shotgun/buckshot
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_SECURITY
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/designs/weapon_designs.dm]: 33,34,36,37

File: code/modules/research/designs/weapon_designs.dm [hunk 5]
```diff
@@ -2,6 +2,60 @@
...
+
+/datum/design/shotgun_dart/sec
+	name = "Shotgun Dart"
+	id = "sec_dart"
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000)
+	build_path = /obj/item/ammo_casing/shotgun/dart
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_SECURITY
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/designs/weapon_designs.dm]: 42,43,45,46

File: code/modules/research/designs/weapon_designs.dm [hunk 6]
```diff
@@ -2,6 +2,60 @@
...
+
+/datum/design/incendiary_slug/sec
+	name = "Incendiary Slug"
+	id = "sec_Islug"
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 2000)
+	build_path = /obj/item/ammo_casing/shotgun/incendiary
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_SECURITY
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/designs/weapon_designs.dm]: 51,52,54,55

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
What you're wanting to do is make them subtypes with the only difference being the items that make it not work so if someone changes the auto design it would "sync" to the proto design. Also shame on you for making it cheaper within the proto undocumented.

**Review by ShizCalev — CHANGES_REQUESTED** [review]
what @ExcessiveUseOfCobblestone said

**Comment by Robustin** [comment]
>Undocumented
>Its in the bloody changelog

**Comment by optimumtact** [comment]
why the random buff at the same time

**Comment by CPTANT** [comment]
Scrap the buff, just merge the functionality.

**Comment by Robustin** [comment]
I wanted to take advantage of splitting the design since I figured the "guns and ammo fabricator" would be more efficient than the public "does everything" fabricator that you have to hack to even unlock most of these designs. 

But if that's holding this up ill change it back when I trim the unnecessary lines.

File: code/modules/research/designs/weapon_designs.dm (hunk 1 above)
Comment (Robustin) [reply code/modules/research/designs/weapon_designs.dm]: I can't remove 7, it needs an ID distinct from its parent

**Comment by Robustin** [comment]
The techfab lost its efficiency bonus for the ammo, its a common sense change though.

**Comment by Bawhoppen** [comment]
I don't think what is common sense to you is common sense to other people.

**Pull request merged**